test/unittests/t_ring_buffer.cc

```diff
@@ -0,0 +1,87 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include <gtest/gtest.h>
+
+#include <cstring>
+
+#include "ring_buffer.h"
+#include "util/pointer.h"
+#include "util/prng.h"
+
+class T_RingBuffer : public ::testing::Test {
+ protected:
+  static const size_t kSize;
+
+  virtual void SetUp() {
+    ring = new RingBuffer(kSize);
+  }
+
+  virtual void TearDown() {
+    ring.Destroy();
+  }
+
+  UniquePtr<RingBuffer> ring;
+};
+
+const size_t T_RingBuffer::kSize = 997;
+
+TEST_F(T_RingBuffer, Basics) {
+  unsigned char buf[kSize];
+  memset(buf, 0, kSize);
+
+  EXPECT_EQ(RingBuffer::kInvalidObjectHandle, ring->PushFront(buf, kSize));
+  EXPECT_EQ(0u, ring->PushFront(buf, kSize - sizeof(size_t)));
+  EXPECT_EQ(0u, ring->free_space());
+  EXPECT_EQ(0u, ring->RemoveBack());
+  EXPECT_EQ(kSize, ring->free_space());
+
+  auto handle_null = ring->PushFront(NULL, 0);
```

Comment:
```suggestion
  ObjectHandle_t handle_null = ring->PushFront(NULL, 0);
```
Sorry, we're not there yet ... :)